apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/RequestResponsePopulateMdcFilter.java

```diff
@@ -42,28 +48,50 @@ public final class RequestResponsePopulateMdcFilter implements Filter {
 
   /** {@inheritDoc} */
   @Override
-  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
-      throws IOException, ServletException {
+  protected void doFilterInternal(
+      HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+      throws ServletException {
+
     /*
      * This should be cleared by Jetty via the request log handling but just in case we have a situation where
      * that handler does not fire (say, due to a Jetty defect) clear it now before the request starts.
      */
     BfdMDC.clear();
+    ContentCachingRequestWrapper reqWrapper = new ContentCachingRequestWrapper(request);
+    ContentCachingResponseWrapper resWrapper = new ContentCachingResponseWrapper(response);
+    // Requests aren't cached until their parameters have been accessed.
+    reqWrapper.getParameterMap();
 
-    handleRequest(request);
+    handleRequest(reqWrapper);
     try {
-      chain.doFilter(request, response);
+      chain.doFilter(reqWrapper, resWrapper);
+      // The original response will not return with its body unless this method is called.
+      resWrapper.copyBodyToResponse();
+    } catch (EOFException e) {
+      /*
+       * The EOFException is a checked exception and is expected when the response body's GZIP stream has reached its end.
+       * We can't control whether external filters or applications continue to read the streamed response body
+       * after its completion, so this is the best we can do.
+       */
+      LOGGER.info("End of stream", e);
+    } catch (IOException e) {
+      /*
+       * The IOException is a checked exception and will be thrown whenever the response body's GZIP stream is interrupted.
+       * This can occur when the ASG is scaling down (like after a load test), or if the client cancels a request mid-stream.
+       * There isn't much we can do on our end aside from catching it.
+       */
+      LOGGER.info("Tried closing stream", e);
```

Comment:
Given the above message this one should probably be a debug or something; we never really need to know this since we can't do much about it.